lib/connection.js

```diff
@@ -112,15 +113,26 @@ class Connection extends EventEmitter {
   }
 
   // Send a handshake packet over the connection
-  //   appName - name of an application to connect to
+  //   app - string or object, application to connect to as 'name' or
+  //         'name@version' or { name, version }
   //   login - user name (optional)
   //   password - user password (optional)
   //   callback - callback function to invoke after the handshake is completed
   //
-  handshake(appName, login, password, callback) {
+  handshake(app, login, password, callback) {
+    let target;
+    if (typeof app === 'string') {
+      const [name, version] = app.split('@');
+      if (semver.valid(version)) target = [name, version];
+      else target = name;
+    } else if (!app.version) {
+      target = app.name;
+    } else {
+      target = [app.name, app.version];
```

Comment:
Can you please stick to either a string (as in `{ handshake: [0, 'app@latest'] }`) or an array (as in `{ handshake: [0, 'app', 'latest'] }`) in both cases? As the parsed messages are plain objects, the current approach may suffer from the constantly changing object shapes.